### PR TITLE
Party Finder Stats improvement

### PR DESCRIPTION
**Pull request by TakoTheDev**
best code

- added mp
- added Gyro
- renamed Dark Claymore to just Claymore
- now scans backpack and ender chest as well
- Fixed Double
- Slight Code cleanup
- added command

### CODE REVIEW

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt [hunk 1]
```diff
@@ -164,61 +164,61 @@ object PartyFinderStats {
...
                         }
+                    val bloodMobsKilled = (profileData.player_stats.kills["watcher_summon_undead"]?.toInt()
+                        ?: 0) + (profileData.player_stats.kills["master_watcher_summon_undead"]?.toInt() ?: 0)
+
+                    component.append(
+                        UTextComponent("§5Miscellanous: §7(Hover)\n\n").setHoverText(
+                            """
+                                #§aTotal Secrets Found: §l§e${NumberUtil.nf.format(secrets)}
+                                #§aBlood Mobs Killed: §l§e${NumberUtil.nf.format(bloodMobsKilled)}
+                                #§dMagical Power: §l§e${profileData.accessory_data.highest_magical_power.toInt()}
```
Comment (jani270) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: Not a big fan of that, too many people just snagged accessories for SB XP and then handed them back.

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: ayo jani270, so ig I need to calculate it? ;-;

Comment (jani270) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: > ayo jani270, so ig I need to calculate it? ;-;

If you want the correct MP, yes

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: Is this accurate currently btw?

Comment (jani270) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: > 
> 
> Is this accurate currently btw?

indeed

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: image
current code (will be pushed once my hypixel-api code gets accepted)

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: Wait wtf what a coincidence, I never saw your message

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: Also note to self, add a command to trigger the message

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: Also last thing for now, do I care about this? "Dungeon Accessories have double the Magical Power while in the Catacombs."

Comment (jani270) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt]: > Also last thing for now, do I care about this? "Dungeon Accessories have double the Magical Power while in the Catacombs."

I would not

**Comment by My-Name-Is-Jeff** [comment]
> best code
> 
> * added mp
> * added Gyro
> * renamed Dark Claymore to just Claymore
> * now scans backpack and ender chest as well
> * Fixed Double
> * Slight Code cleanup
> * added command

half these changes don't actually exist sooooo

**Comment by TakoTheDev** [comment]
Yea I'm waiting for the hypixel-api thing to be pulled so the build doesn't fail and I have to re-commit something

**Comment by My-Name-Is-Jeff** [comment]
> Yea I'm waiting for the hypixel-api thing to be pulled so the build doesn't fail and I have to re-commit something

Just push it

**Comment by TakoTheDev** [comment]
ig will do in like ~6 hours?

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt [hunk 2]
```diff
@@ -165,64 +175,110 @@ object PartyFinderStats {
...
+                                ).ifBlank { "§c§lNone" })
                             )
                         }
+                    val bloodMobsKilled = (profileData.player_stats.kills["watcher_summon_undead"]?.toInt()
+                        ?: 0) + (profileData.player_stats.kills["master_watcher_summon_undead"]?.toInt() ?: 0)
+
+                    val magicalPower = profileData.inventory.bag.talisman.toMCItems().filterNotNull().map {
+                        val itemId = ItemUtil.getSkyBlockItemID(it)?.let { id ->
+                            when {
+                                id.startsWith("PARTY_HAT_") -> "PARTY_HAT"
```
Comment (TakoTheDev) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt:189]: Would probably be smart to have a map with ids and what they should be replaced with.

File: src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt [hunk 3]
```diff
@@ -47,6 +48,7 @@ object PartyFinderStats {
     private val partyFinderRegex = Regex(
         "^Party Finder > (?<name>\\w+) joined the dungeon group! \\((?<class>Archer|Berserk|Mage|Healer|Tank) Level (?<classLevel>\\d+)\\)$"
     )
+    private val requiredRegex = Regex("§7§4☠ §cRequires §5.+§c.")
```
Comment (TakoTheDev) [inline src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt:51]: it seems that there's also a requirement nbt tag (for example: dungeon_skill_req:"CATACOMBS:14") so maybe use that

Comment (TakoTheDev) [reply src/main/kotlin/gg/skytils/skytilsmod/features/impl/dungeons/PartyFinderStats.kt:51]: Actually forget that, it exists for all items even if you have the requirement

**Pull request merged**